tests/cpp/test_multidevice_communications.cpp

```diff
@@ -246,13 +245,15 @@ TEST_P(CommunicationTest, SendRecvToSelf) {
   at::Tensor input_tensor = at::empty({kTensorSize}, tensor_options);
   at::Tensor output_tensor = at::empty_like(input_tensor);
 
+  c10d::Backend* backend =
+      communicator->getBackendForTeam(communication->team(), GetParam());
   for (auto repetition : c10::irange(kNumRepetitions)) {
     input_tensor.copy_(at::arange(kTensorSize, tensor_options) + repetition);
 
     postSingleCommunication(
         communication,
         communicator->deviceId(),
-        backend_,
+        backend,
```

Comment:
both backend are valid, and `backend_` (i.e. the world communicator) is probably more representative of how we are going to use `SendRecv`

Reply:
Agreed. I changed the PR description to not say "right" because both are right. This change is only to expose the bug.